extensions/default/src/DicomWebDataSource/utils/clientManager.ts

```diff
@@ -0,0 +1,162 @@
+import { api } from 'dicomweb-client';
+import StaticWadoClient from './StaticWadoClient';
+import dcm4cheeReject from '../dcm4cheeReject';
+import { errorHandler, utils } from '@ohif/core';
+
+export default class clientManager {
+  clients;
+  userAuthenticationService;
+
+  // adds a dicomweb server configuration in the clients list
+  private _addConfiguration(givenConfig) {
+    const config = Object.assign({}, givenConfig);
+    config.qidoConfig = {
+      url: config.qidoRoot,
+      staticWado: config.staticWado,
+      singlepart: config.singlepart,
+      headers: this.userAuthenticationService.getAuthorizationHeader(),
+      errorInterceptor: errorHandler.getHTTPErrorHandler(),
+    };
+
+    config.wadoConfig = {
+      url: config.wadoRoot,
+      staticWado: config.staticWado,
+      singlepart: config.singlepart,
+      headers: this.userAuthenticationService.getAuthorizationHeader(),
+      errorInterceptor: errorHandler.getHTTPErrorHandler(),
+    };
+
+    config.qidoDicomWebClient = config.staticWado
+      ? new StaticWadoClient(config.qidoConfig)
+      : new api.DICOMwebClient(config.qidoConfig);
+
+    config.wadoDicomWebClient = config.staticWado
+      ? new StaticWadoClient(config.wadoConfig)
+      : new api.DICOMwebClient(config.wadoConfig);
+
+    config.qidoDicomWebClient.name = config.name;
+    config.wadoDicomWebClient.name = config.name;
+    this.clients.push(config);
+  }
+
+  public addConfiguration(params, query, config) {
+    if (
+      config.onConfiguration &&
+      typeof config.onConfiguration === 'function'
+    ) {
+      config = config.onConfiguration(config, {
+        params,
+        query,
+      });
+    }
+    this._addConfiguration(config);
+  }
+
+  public getAuthorizationHeader() {
```

Comment:
Does this need to be public? I think it is currently only referenced from within this class.